### ISSUE TITLE
Regarding the Bernoulli Mask for Unlabeled Data

### ISSUE THREAD

**Issue by shreejalt** [LeeHyuck/ABCtorch#3]
Hi, 
Thanks for your work. I had one doubt regarding the bernoulli distribution of the unlabeled dataset.
Can you point out where do you calculate the B(N_L/N_qhat ) in the code.

It seems you only use the statistics of the labeled distribution `ir2` in the formula `1 - (epoch/500) * (1-ie2)`

Thanks

**Comment by LeeHyuck** [comment]
Hello thank you for reading my paper

In our paper we assumed that the labeled set and unlabeled set share same class distribution

In the code B(N_L/N_qhat ) was calculated as&nbsp;

label_u = torch.zeros(batch_size, num_class).scatter_(1, label_u.cpu().view(-1, 1), 1)maskforbalanceu = torch.bernoulli(torch.sum(label_u.cuda(0) * torch.tensor(ir22).cuda(0), dim=1).detach())Thanks
-----Original Message-----
From: "Silver10" ***@***.***&gt;
To: "LeeHyuck/ABC" ***@***.***&gt;;
Cc: "Subscribed" ***@***.***&gt;;
Sent: 2022-05-13 (금) 22:22:38 (UTC+09:00)
Subject: [LeeHyuck/ABC] Regarding the Bernoulli Mask for Unlabeled Data (Issue #3)


Hi,Thanks for your work. I had one doubt regarding the bernoulli distribution of the unlabeled dataset.Can you point out where do you calculate the B(N_L/N_qhat ) in the code.
It seems you only use the statistics of the labeled distribution ir2 in the formula 1 - (epoch/500) * (1-ie2)
Thanks
